src/peer_digest.cc

```diff
@@ -167,6 +167,8 @@ peerDigestDestroy(PeerDigest * pd)
 
 PeerDigest::~PeerDigest()
 {
+    if (times.next_check)
```

Comment:
This eventDelete() call will crash Squid if the event callback has been already moved from the event.cc queue into AsyncQueue. Look for debug_trap() in EventScheduler::cancel(). A proper fix requires event API improvements (IIRC there is an unfinished official PR about that). For now, since this code is not reached frequently, I think we can afford to check (instead of potentially accumulating these events):

```suggestion
    if (times.next_check && eventFind(peerDigestCheck, this))
```

Reply:
Done.
Note that some existing `eventDelete()` callers do not check with `eventFind()`, e.g.,  `TunnelStateData::Connection::~Connection()`.

Reply:
Yes, all unconditional eventDelete() callers that supply a non-nil second argument are buggy IIRC.